packages/react-server/core/ClientController.js

```diff
@@ -650,7 +650,11 @@ class ClientController extends EventEmitter {
 
 			_.forEach(
 				getRootElementAttributes(element),
-				(v, k) => root.setAttribute(k, v)
+				(v, k) => {
+					if (k !== 'tagName') {
+						root.setAttribute(k, v)
+					}
+				}
```

Comment:
Yea, its a kinda ugly bit of code.  No harm arises by putting the attribute on the node, but its redundant information because the node with be of the type specified.